### ISSUE TITLE
Improve TypeScript coverage of Redux store in fxa-payments-server

### ISSUE THREAD

**Issue by lmorchard** [mozilla/fxa#1079]
I kind of fudged the TypeScript annotations in the Redux store code for fxa-payments-server with very broad types. It could be reworked to be much more specific - which would enable much better auto-complete in IDEs and validate & lint action names and parameters in code.

**Comment by lmorchard** [comment]
FWIW, [I started tinkering](https://github.com/lmorchard/fxa/commit/6892067e6de88643b656a951ff61990555861e46) with reworking the Redux store toward resolving this issue. But, putting this work on pause for other priorities.

**Comment by lmorchard** [comment]
To elaborate on an aspect that could be improved here:

Actions in the Redux store are [defined using `createActions()` from the redux-actions module](https://github.com/mozilla/fxa/blob/master/packages/fxa-payments-server/src/store/index.tsx#L72). That's [a utility](https://redux-actions.js.org/api/createaction#createactions) which does a lot of magical work to generate Redux action creators while reducing boilerplate. The problem, though, is that this magic doesn't play well with TypeScript and obscures types.

As a practical example - if you edit a React component that imports `actions` from the `store/index.tsx` module, there's no auto-complete of action names in VSCode. That's because the list of available actions are obscured on the way through `createActions` and I have not found a good set of type definitions that fixes this.

So, one solution is to create a new `actions.ts` module under `store` and split each of these actions out into a function explicitly exported from that module.

This might introduce the dreaded boilerplate, which could be okay as a cost of making the Redux store more self-documenting when used in other places in the app. Would be nice if we could keep the repeated code to a minimum while improving type coverage, though.

**Comment by lmorchard** [comment]
There are three big niceties from `createAction` in `redux-actions`. 

* Giving every action creator function a `.toString()` method, so we don't need a separate list of `ACTION_TYPE` constants
* Building [Flux action objects](https://github.com/redux-utilities/flux-standard-action) with minimal boilerplate.
* Generating identity function action creators with just a list of names.

Consider this code:
```
const actions = createActions(
  { foo: param => `foo ${param}` },
  "bar"
);
```
This is semi-equivalent to:
```
const actions = {};

actions.foo = param => ({
  type: 'foo',
  payload: `foo ${param}`,
});

actions.foo.toString = () => 'foo';

actions.bar = param => ({
  type: 'bar',
  payload: param,
});

actions.bar.toString = () => 'bar';
```
And, the `.toString()` method comes in handy when [defining reducers](https://github.com/mozilla/fxa/blob/master/packages/fxa-payments-server/src/store/index.tsx#L245).

**Comment by lmorchard** [comment]
Lower priorities, but maybe also important: The single `index.tsx` in `store` could & should probably be broken up into separate smaller modules for selectors, actions, thunks, and reducers.

For bonus points, we should backfill tests to improve coverage of the Redux store - which is essentially *not* covered by tests at all right now

**Comment by LZoog** [comment]
Fixed in #2598
